File: algorithms/204.cpp
#include <stdio.h>
#include <vector>
using namespace std;

class Solution
{
public:
    int countPrimes(int n)
    {
        vector<bool> isPrime(n, true);
        vector<int> prime;
        for ( int i = 2; i < n; i++ )
        {
            if ( isPrime[i] )
                prime.push_back(i);
            for ( auto it = prime.begin(); it != prime.end(); it++ )
            {
                if ( *it * i >= n )
                    break;
                isPrime[*it*i] = false;
                if ( i % *it == 0 )
                    break;
            }
        }
        return prime.size();
    }
};

int main()
{
    return 0;
}

File: algorithms/python/204.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        isPrime = [1] * n
        isPrime[:2] = [0, 0]
        for i in range(2, n):
            if isPrime[i]:
                for j in range(2*i, n, i):
                    isPrime[j] = 0
        return sum(isPrime)
